jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -83,5 +86,48 @@ def process(self, input, *args, **kwargs):
             slit_y_range = [self.slit_y_low, self.slit_y_high]
             result = load_wcs(input_model, reference_file_names, slit_y_range)
 
+            if sip_approx:
+                if degree is not None:
+                    if degree > 6:
+                        raise ValueError('Degree for SIP approximation must be less or equal to 6.')
+                if inv_degree is not None:
+                    if inv_degree > 6:
+                        raise ValueError('Degree for SIP approximation must be less or equal to 6.')
+                # check if its imaging mode
+                if result.meta.exposure.type.lower() in IMAGING_TYPES:
+
+                    # fit sip approx., degree is chosen by best fit
+                    try:
+                        fit_sip_hdr = result.meta.wcs.to_fits_sip(None, max_pix_error,
+                                      degree, max_inv_pix_error, inv_degree,
+                                      npoints)
+                    except ValueError:
+                        return result
+
+                    # check if polynomial terms were actually fit, or just linear.
+                    sip = False or ('SIP' in fit_sip_hdr['ctype1'])
+
+                    # if best fit > 6th degree, fit a 6th degree instead.
+                    # and give up if that doesn't work.
+                    if sip:
+                        if fit_sip_hdr['a_order'] > 6:
+                            try:
+                                fit_sip_hdr = result.meta.wcs.to_fits_sip(None, max_pix_error,
+                                              6, max_inv_pix_error, 6, npoints)
```

Comment:
Even though it is not likely, in principle it is possible that the degree for the inverse polynomial be smaller than for the forward one. The logic here forces inverse to be 6 (even though it might be 5) if forward's degree is > 6.

NOTE: I do not think this is critical to be fixed.

In addition, I wish, as suggested in https://github.com/spacetelescope/gwcs/pull/286#issuecomment-592650047, `degree` could be a list of degrees that `to_fits_sip()` would consider for fitting. Then this logic here would not be necessary at all if you could set `degree=[2,3,4,5,6]`. CC: @perrygreenfield

Reply:
Unless the accuracy for the inverse is to be poor, I have a hard time thinking of a case where it would be lower.

Reply:
I do not disagree with that.

But things would be way easier/simpler if you would allow `degree` to be a list in `gwcs`.

Reply:
⬆️

Reply:
i'm going to leave this unresolved, it seems like this could be an improvement later?

Reply:
@cshanahan1 Take a look at https://github.com/spacetelescope/jwst/pull/5507#pullrequestreview-549584346 That fixes, I believe, the first part + another issue described in the comment - you do not detect the situation when inverse degree is over 6 but forward is `<=6`.

The other issue with simplification of the logic - that one requires changes in `gwcs` and it cannot be done now.